### PR TITLE
feat: add DoubleMd5UserSaltCredManager

### PR DESCRIPTION
**Pull request by luliangce**
When I migrated the user system from an old service to Casdoor and synchronized the users, I found that there was no suitable password algorithm. This algorithm may be relatively obscure, but I hope to add it to the Casdoor.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@luliangce can you share which famous systems or services use this? Like: https://casdoor.org/docs/organization/overview/

image

File: cred/manager.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ func GetCredManager(passwordType string) CredManager {
 		return NewPbkdf2SaltCredManager()
 	} else if passwordType == "argon2id" {
 		return NewArgon2idCredManager()
+	} else if passwordType == "double-md5-salt" {
```
Comment (hsluoyz) [inline cred/manager.go:37]: Put after "md5-salt"

File: web/src/OrganizationEditPage.js [hunk 2]
```diff
@@ -257,7 +257,7 @@ class OrganizationEditPage extends React.Component {
           </Col>
           <Col span={22} >
             <Select virtual={false} style={{width: "100%"}} value={this.state.organization.passwordType} onChange={(value => {this.updateOrganizationField("passwordType", value);})}
-              options={["plain", "salt", "sha512-salt", "md5-salt", "bcrypt", "pbkdf2-salt", "argon2id"].map(item => Setting.getOption(item, item))}
+              options={["plain", "salt", "sha512-salt", "md5-salt", "bcrypt", "pbkdf2-salt", "argon2id", "double-md5-salt"].map(item => Setting.getOption(item, item))}
```
Comment (hsluoyz) [inline web/src/OrganizationEditPage.js:260]: Put after "md5-salt"

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@luliangce any update?

**Comment by luliangce** [comment]
> @luliangce can you share which famous systems or services use this? Like: https://casdoor.org/docs/organization/overview/
> 
> image

After thinking it over again, I don't think any popular services are using this, so I've decided this change isn't really  good  for merging into casdoor.

**Pull request closed without merging**